lib/mpl_toolkits/axes_grid1/axes_divider.py

```diff
@@ -512,6 +516,8 @@ def append_axes(self, position, size, pad=None, add_to_figure=True, *,
         **kwargs
             All extra keywords arguments are passed to the created axes.
         """
+        _api.check_in_list(["left", "right", "bottom", "top"],
+                           position=position)
```

Comment:
FWIW if you're going to change this block I tend to like writing these as
```python
        create_axes, pack_start = _api.check_getitem({
            "left": (self.new_horizontal, True),
            "right": (self.new_horizontal, False),
            "bottom": (self.new_vertical, True),
            "top": (self.new_vertical, False),
        }, position=position)
        ax = create_axes(
            size, pad, pack_start=pack_start, axes_class=axes_class, **kwargs)
```
which makes the exhaustiveness check clearer.

Reply:
Sure, I didn't really consider the content of the if-clauses, just realized that the check was at the bottom...